nixos/modules/services/web-apps/miniflux.nix

```diff
@@ -0,0 +1,130 @@
+      [ "$(${pgsu} ${pgbin}/psql -Atc "select 1 from pg_database where datname='$1'")" == "1" ]
+    }
+    if ! db_exists "${dbName}"; then
+      ${pgsu} ${pgbin}/psql postgres -c "CREATE ROLE ${dbUser} WITH LOGIN NOCREATEDB NOCREATEROLE ENCRYPTED PASSWORD '${dbPassword}'"
+      ${pgsu} ${pgbin}/createdb --owner "${dbUser}" "${dbName}"
+      ${pgsu} ${pgbin}/psql "${dbName}" -c "CREATE EXTENSION IF NOT EXISTS hstore"
+    fi
+
+    if [ -z "${cfg.adminCredentialsFile}" ]; then
+      cat > /run/miniflux/admin-credentials <<- EOF
+    	ADMIN_USERNAME=admin
+    	ADMIN_PASSWORD=password
+    	EOF
+      chown "${cfg.user}:${cfg.group}" /run/miniflux/admin-credentials
+      chmod 600 /run/miniflux/admin-credentials
+    fi
+  '';
+in
+
+{
+  options = {
+    services.miniflux = {
+      enable = mkEnableOption "miniflux";
+
+      user = mkOption {
+        type = types.str;
+        default = "miniflux";
+        description = ''
+          User account under which miniflux runs.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "miniflux";
+        description = ''
+          User group under which miniflux runs.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = ''
+          The host the daemon will listen on.
+        '';
+      };
+
+      listenPort = mkOption  {
+        type = types.int;
+        default = 8080;
+        description = ''
+          The port the daemon will listen on.
+        '';
+      };
+
+      adminCredentialsFile = mkOption  {
+        type = types.str;
+        default = "";
+        description = ''
+          File containing the ADMIN_USERNAME and ADMIN_PASSWORD (length >= 6),
+          in the format of an EnvironmentFile as described by systemd.exec(5).
+        '';
+        example = "/etc/nixos/miniflux-admin-credentials";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.postgresql.enable = true;
+
+    users = {
+      users.miniflux = {
+        name = cfg.user;
+        group = cfg.group;
+        description = "miniflux service user";
+        isSystemUser = true;
+        createHome = false;
+      };
+      groups.miniflux = {
+        name = cfg.group;
+      };
+    };
+
+    systemd.services.miniflux = {
+      description = "Miniflux service";
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "postgresql.service" ];
+      after = [ "network.target" "postgresql.service" ];
+
+      serviceConfig = {
+        ExecStart = "${pkgs.miniflux}/bin/miniflux";
+        ExecStartPre = "+${preStart}";
+        User = cfg.user;
+        Group = cfg.group;
+        RuntimeDirectory = "miniflux";
+        RuntimeDirectoryMode = "0700";
+        EnvironmentFile = [
+          "${cfg.adminCredentialsFile}"
+          "-/run/miniflux/admin-credentials"
+        ];
+      };
+      unitConfig.JoinsNamespaceOf = "postgresql.service";
```

Comment:
Good job on the changes! I don't believe this is required anymore, but I could be wrong about that.

Reply:
You're right, I seems to have missed this part of the docs:
> If the executable path is prefixed with "+" [...] the various file system namespacing options (such as PrivateDevices=, PrivateTmp=) are not applied to the invoked command line [...]
Thank you for your help on this PR, I learned quite a bit in the process.

Reply:
As a final request... if join namespace of is no longer required I think you can use dynamic user which would be great.

Reply:
You have your eyes everywhere; it's done,